test/ecto/integration/math_test.exs

```diff
@@ -89,7 +89,7 @@ defmodule Ecto.Integration.MathTest do
 
       query = from(p in Product, select: fragment("acos(?)", p.price))
 
-      [nil] = TestRepo.all(query)
+      assert [nil] = TestRepo.all(query)
```

Comment:
You don't have to assert the match here because if it doesn't match an error will be thrown.

Reply:
True, but you don't get proper highlighting and notice about what was actually returned without having those.

Reply:
Odd because the error about the match will fire before the assert happens. Unless assert is a macro that manipulates the AST tree.

Reply:
`assert` is a macro, and quite an intelligent one. E.g. if you do `assert Decimal.eq?(decimal1, decimal2)` it'll give you the values of both `decimal1` and `decimal2` if the assertion doesn't hold.

Reply:
Hmm didn't know that. Always felt redundant to assert a match. Carry on then.